modules/product/src/main/java/com/opengamma/strata/product/etd/EtdStyleType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.etd;
+
+import org.joda.convert.FromString;
+import org.joda.convert.ToString;
+
+import com.google.common.base.CaseFormat;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * The type of an Exchange Traded Derivative (ETD) product, either a future or an option.
+ */
+public enum EtdStyleType {
```

Comment:
@joburnham, @phuvan and I have been discussing this and have a couple of alternatives that avoid `StyleType`

* Rename `EtdStyle` to `EtdInfo` and `EtdStyleType` to `EtdStyle`
* Rename `EtdStyleType` to `EtdVariant`